### PR TITLE
PR for #12352: Default annotations for kwargs inited to constants

### PR DESCRIPTION
**Pull request by edreamleo**
PR for mypy issue #12352: Don't require annotations of kwargs initialized to bool, float, int or string constants.

### New and changed files

- mypy/fastparse.py: make_argument creates "virtual" annotations from constant initializers.
- test-data/unit/check-default-annotations.test: New unit tests.
- ekr_test.py: (temporary) A simple test file for running mypy without unit testing framework.
- mypy/test/testcheck.py: Add a line for check.default-annotations.test.
- [mypy.leo](https://github.com/leo-editor/leo-editor-contrib/blob/master/StudyOutlines/mypy.leo): A [Leo](http://leoeditor.com/) outline containing almost all of mypy's sources.

### Test Plan

- Add unit tests in test-data/unit to test all changed annotations.
- Verify that semantic analysis and type checking handle default annotations as expected.
- Many unit tests may (properly) change as the result of the PR. Update these tests.

We expect some unit tests to fail with the new code!  Here are the new failures:
```console
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-classes.test::testInitSubclassUnannotatedMulti
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-fastparse.test::testFastParsePerArgumentAnnotations
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-fastparse.test::testFastParsePerArgumentAnnotationsWithReturn
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-functions.test::testEllipsisWithArbitraryArgsOnBareFunctionWithDefaults
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-columns.test::testColumnsMethodDefaultArgumentsAndSignatureAsComment
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-functions.test::testDefaultArgumentsAndSignatureAsComment
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-functions.test::testMethodDefaultArgumentsAndSignatureAsComment
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-attr.test::testAttrsUsingBadConverter
FAILED mypy/test/testcheck.py::TypeCheckSuite::check-attr.test::testAttrsUsingBadConverterReprocess
FAILED mypy/test/testparse.py::ParserSuite::parse.test::testDefaultArgs
FAILED mypy/test/testparse.py::ParserSuite::parse.test::testBareAsteriskInFuncDef
FAILED mypy/test/teststubgen.py::StubgenPythonSuite::stubgen.test::testDefaultArgInt
FAILED mypy/test/teststubgen.py::StubgenPythonSuite::stubgen.test::testDefaultArgBool
FAILED mypy/test/teststubgen.py::StubgenPythonSuite::stubgen.test::testDefaultArgStr
FAILED mypy/test/teststubgen.py::StubgenPythonSuite::stubgen.test::testDefaultArgBytes
FAILED mypy/test/teststubgen.py::StubgenPythonSuite::stubgen.test::testDefaultArgFloat
FAILED mypy/test/teststubgen.py::StubgenPythonSuite::stubgen.test::testVarArgsWithKwVarArgs
FAILED mypy/test/teststubgen.py::StubgenPythonSuite::stubgen.test::testKeywordOnlyArg
```
I'll investigate these 18 new failures and modify the test cases as appropriate.

### CODE REVIEW

**Comment by edreamleo** [comment]
Sheesh. The bot's complaints are pure cruft. I'd like to minimize the report, at least.

File: mypy/fastparse.py [hunk 1]
```diff
@@ -979,11 +1003,23 @@ def make_argument(self, arg: ast3.arg, default: Optional[ast3.expr], kind: ArgKi
             arg_type = None
             if annotation is not None:
                 arg_type = TypeConverter(self.errors, line=arg.lineno).visit(annotation)
+            elif isinstance(default, ast3.Constant):  # #12352.
```
Comment (JelleZijlstra) [inline mypy/fastparse.py:1006]: We should probably do this only *after* trying to find the type in a type comment.

File: mypy/fastparse.py [hunk 2]
```diff
@@ -979,11 +1003,23 @@ def make_argument(self, arg: ast3.arg, default: Optional[ast3.expr], kind: ArgKi
...
+                trace = True
+                arg_class = default.value.__class__.__name__
+                if arg_class in ('bool', 'bytes', 'float', 'int', 'str'):
+                    arg_type = UnboundType(arg_class)
```
Comment (JelleZijlstra) [inline mypy/fastparse.py:1010]: I believe this will behave incorrectly if the name is shadowed in the local namespace.

Comment (edreamleo) [reply mypy/fastparse.py:1010]: Thanks for your comment! I'll see what I can do.

**Pull request closed without merging**